### ISSUE TITLE
SteamVR_ControllerManager has been removed without a replacement

### ISSUE THREAD

**Issue by Cluster7ck** [ValveSoftware/steamvr_unity_plugin#125]
There is no SteamVR_ControllerManager in the new update and I couldn't find any equivalent script.
I'm not sure how to manage extra Vive trackers now. Is there a way to do that, that I haven't seen?
I tried using the old SteamVR_ControllerManager script, but the behavior definitely changed and I think the device IDs get incorrectly assigned. I placed some primitives on my SteamVR_TrackedObjects and when you hit play the primitives and the controllers either occupy the same space (same transform) or the tracked objects replace the controller entirely.

**Comment by bangjieding** [comment]
I have the same problem. Finally I use last version

**Comment by Cluster7ck** [comment]
Hm. I think you have to go through the new Action system and make pose actions for each tracker. What I think is kinda weird though is that you have to assign a fix role to trackers, like Waist, Left Foot, Chest, etc.

I'm going to close this Issue when I can reproduce the previous behavior, but for now I'm leaving it open for discussion.

**Comment by Cluster7ck** [comment]
What I got so far:
For every tracker you want to use, create a pose action in the Action Set you are using (you can do this in the SteamVR Input Window). Then you open the binding UI. There you have to click on whatever _Current Controller_ is selected and click on the tracker in the popup. Now the selected controller should be the tracker and you can go into _MANAGE TRACKERS_ and assign a role for each tracker you have. I have no idea why there are distinct roles and why there are less roles than the possible amount of trackers you can connect. After you have assigned the roles you can select the trackers in the _Current Controlelr_ popup and then click on _Create Binding_. There you click on _Edit Action Poses_ and on the left under _POSES:_ click on _Unused_ and select one of the pose actions you have created for your trackers. Go back and select _Vive Controller_ for the _Current Controller_ and click on Edit (the binding). There you have to click _Replace Default Binding_ and save. Now your `actions.json` and `bindings_vive_controller.json` should contain all the necessary references.

Then go back to unity and click on _Save and generate_. Create gameobjects for your trackers and add the `SteamVR_Behaviour_Pose` script. In the _Pose Action_ dropdown you can now select one of the actions you created.

After doing all of this it still doesn't work and I have no idea why. When I hit play I see that the default bindings are selected but the tracker object and the controller are still in the same place.

**Comment by Trecanius** [comment]
Got the Vive trackers to work after manually editing the .json files.
For each tracker, create a pose action and a assign a role in the binding UI, following @Cluster7ck steps.
You should have a binding .json for each tracker you want to use. Here's one for a left foot tracker:
```
{
   "app_key" : "application.generated.unity.steamvr.exe",
   "bindings" : {
      "/actions/default" : {
         "chords" : [],
         "haptics" : [],
         "poses" : [
            {
               "output" : "/actions/default/in/poseleftfoot",
               "path" : "/user/foot/left/pose/raw"
            }
         ],
         "skeleton" : [],
         "sources" : []
      }
   },
   "controller_type" : "vive_tracker_left_foot",
   "description" : "",
   "name" : "vive_tracker_left_foot"
}
```
Make sure `app_key` is set to `application.generated.unity.steamvr.exe` in your binding .json files.
In `actions.json` set the default bindings: 
```
"default_bindings": [
    {
      "controller_type": "vive_controller",
      "binding_url": "bindings_vive_controller.json"
    },
    {
      "controller_type": "vive_tracker_left_foot",
      "binding_url": "bindings_vive_tracker_left_foot.json"
    },
    {
      "controller_type": "vive_tracker_right_foot",
      "binding_url": "bindings_vive_tracker_right_foot.json"
    }
  ]
```

**Comment by zite** [comment]
The vive tracker UI has received an update now and you should be able to bind actions to specific trackers now. Let me know if this is still an issue and I'll reopen.

**Comment by AndreTInfante** [comment]
For what it's worth, I'm having the same issue as Cluster7, using the latest version of SteamVR and the UI. I've created a pose action for every tracker, and bound them using the UI, and over-written the default binding with the new one, but they don't move from (0,0,0). Is there a way to fix it without manually editing json?

**Comment by Cluster7ck** [comment]
@keithbradner I came back to this because I thought 2.2.0 might have fixed the problems I had the first time around, but the problems seem to be still there.

It's very hard to give a clear description of the problem, because random things completely change how the binding menu behaves.

I got it to work for one tracker with some hassle. First it didn't move from (0,0,0), when I hit play in the editor so I made a build and that fixed it in the build AND the editor(???). 
Then I tried to add more trackers. So i created additional pose actions for each. Including the first I created, I now have 5 extra pose actions. I go into the binding UI and try to assign the actions to the trackers. when I click on the action nothing happens. I look into the developer console of chrome and it throws an error every time I try to assign a pose action to a tracker: `SetActionForPose: Invalid action set specified`. I remember it having a weird behavior when Steam isn't open. So I open Steam and start play mode. I get prompted by the overlay and assign the actions. They are now saved for my steam profile and not as defaults. Also only some of the trackers move (I suspect it gets confused between steam and the defaults). 

Also how am I supposed to save it as default when I assign a pose action? I can click  `Replace Default Binding` and that generates a new .json with the tracker role in the name. So now I have 5 new .jsons, of which maybe one contains all my pose bindings. I tried manually writing one json and then making a build, but then the overlay prompts me and tells me it couldn't load a default binding. 

As I said this isn't the best description of the problems and probably not all of what I did in the right order, but when building and having steam open changes the behavior it's really hard to reproduce bugs. Also the bug where the ui throws  `Invalid action set specified` doesn't always happen when steam is not open and never when steam is open.
All in all I'm very confused. I would make a video of what is happening, but that is kinda hard with half of the interesting stuff taking place in the HMD.
